compiler/rustc_middle/src/ty/context.rs

```diff
@@ -1376,6 +1376,9 @@ pub struct GlobalCtxt<'tcx> {
     /// Common consts, pre-interned for your convenience.
     pub consts: CommonConsts<'tcx>,
 
+    /// A cache for the `erase_regions` function.
+    pub(in crate::ty) erased_region_cache: ShardedHashMap<Ty<'tcx>, Ty<'tcx>>,
```

Comment:
Could you add comments explaining why 1) the query cache is not used here and 2) a sharded hash map is used instead of a simple one.